src/main/java/thanos/commands/DeleteCommand.java

```diff
@@ -0,0 +1,62 @@
+package thanos.commands;
+
+import thanos.exceptions.InvalidCommandException;
+import thanos.tasks.Task;
+import thanos.tasks.TaskList;
+
+/**
+ * Represents a command to delete a task from the {@code TaskList}.
+ * It expects an index of the task to be deleted, which is used to locate and remove the task from the list.
+ */
+public class DeleteCommand extends Command {
+    /**
+     * Constructs a {@code DeleteCommand} with the given argument.
+     *
+     * @param argument the argument associated with this command. The argument should be a string representing
+     *                 the index of the task to be deleted. The index is expected to be an integer value.
+     */
+    public DeleteCommand(String argument) {
+        super(argument);
+    }
+
+    /**
+     * Executes the command to delete a task from the {@code TaskList}.
+     * <p>
+     * This method parses the command argument to obtain the task index, verifies the format and index validity,
+     * removes the task from the {@code TaskList}, and returns a formatted string confirming the removal of the task
+     * and displaying the updated task count. If the argument is empty, incorrectly formatted, or if the index is
+     * invalid, an {@code InvalidCommandException} is thrown.
+     * </p>
```

Comment:
It would be nicer if the description would be less verbose and the details be placed as comments within the code itself, so that it will be easy for others to follow.